### PR TITLE
[#IOPID-857] lock session endpoint

### PR DESCRIPTION
**Pull request by arcogabbo**
This PR depends on https://github.com/pagopa/io-backend/pull/1056

#### List of Changes

* created `lock-session` endpoint

#### Motivation and Context

introduce new endpoint

#### How Has This Been Tested?

unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that all handlers have their respective function declaration file `function.json`

### CODE REVIEW

File: src/utils/ioweb/dependency.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import { Client } from "../../generated/definitions/backend-internal/client";
+
+export type LockSessionDependency = {
+  readonly backendInternalClient: Client<"token">;
+};
```
Comment (BurnedMarshal) [inline src/utils/ioweb/dependency.ts]:
```suggestion
export type LockSessionDependency = LogoutDependencies;
```

File: src/functions/__tests__/lock-session.test.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import { httpHandlerInputMocks } from "../__mocks__/handlerMocks";
+import * as H from "@pagopa/handler-kit";
+import { aFiscalCode } from "../__mocks__/general";
+import * as E from "fp-ts/lib/Either";
+
+const aValidBody = { fiscal_code: aFiscalCode, unlock_code: "123456789" };
+
+const mockLockUserSession = jest
+  .fn()
+  .mockResolvedValue(E.right({ status: 201 }));
```
Comment (BurnedMarshal) [inline src/functions/__tests__/lock-session.test.ts]: If the API returns a `no content` maybe the status code should be `204`

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart minor changes

File: src/functions/__tests__/lock-session.test.ts (hunk 2 above)
Comment (arcogabbo) [reply src/functions/__tests__/lock-session.test.ts]: addressed in https://github.com/pagopa/io-functions-fast-login/pull/19/commits/9a948e0923492e6d126189f32f5fcc5cd4f8ed8a

**Pull request merged**